ops/model.py

```diff
@@ -183,6 +183,14 @@ def is_leader(self):
                 'cannot determine leadership status for remote applications: {}'.format(self)
             )
 
+    def set_workload_version(self, version):
```

Comment:
We could make this a property if we prefer:
```
Unit.workload_version = 'a.b.c'
```

However, there is no 'application-version-get' so you couldn't do 'Unit.workload_version' which would feel weird. If we really want to, we could save it in the framework db, but I feel the use case really is for an output variable, and having just a method instead of an attribute makes it clear you can't read it.

Reply:
yeah, a property you can't read is weird, so +1 to not exposing it that way